### ISSUE TITLE
ES6 class functions triggers false positive for space-before-function-parentheses rule.

### ISSUE THREAD

**Issue by jannisg** [eslint/eslint#2128]
I'm using ESLint version: `0.16.2`
Rule: `"space-before-function-parentheses": [1, "always"]`
Message: `10:17  warning  Missing space before function parentheses  space-before-function-parentheses`

On this code (and all ES6 class functions):

```
class DynamicTabsController {

    constructor ($location, $timeout) {

        console.log('DynamicTabsController :: Init.', this);

        this.$location = $location;
        this.$timeout = $timeout;
        this.tabs = [];
        this.selectedIndex = 0;

    }

    updateTab (options) {
        …
    }
    …
```

Both `constructor` and `updateTab` will trigger the message for having a space between the name and the parens even though the above should be the correct format.

----

`.eslintrc` is below:

```
{
  "parser": "babel-eslint",
  "env": {
    "browser": true,
    "es6": true,
    "node": true,
    "amd": true
  },
  "globals": {
    "angular": false,
    "system": false
  },
  "ecmaFeatures": {
    "modules": true,
    "classes": true,
    "blockBindings":true
  },
  "rules": {
    "curly": [2, "all"],
    "eqeqeq": [2, "smart"],
    "no-eval": 2,
    "no-extend-native": 2,
    "no-extra-bind": 2,
    "no-implied-eval": 2,
    "no-iterator": 2,
    "no-multi-spaces": 2,
    "no-multi-str": 2,
    "no-redeclare": 2,
    "no-sequences": 2,
    "no-with": 2,
    "radix": 2,
    "strict": 0,
    "no-delete-var": 2,
    "no-shadow-restricted-names": 2,
    "no-undef": 0,
    "no-unused-vars": 1,
    "no-use-before-define": [2, "nofunc"],

    "indent": 4,
    "brace-style": [2, "1tbs", { "allowSingleLine": true }],
    "camelcase": 2,
    "comma-spacing": [2, { "before": false, "after": true }],
    "consistent-this": [1, "vm"],
    "eol-last": 1,
    "func-style": 0,
    "key-spacing": [2, { "beforeColon": false, "afterColon": true }],
    "new-cap": [2, { "newIsCap": true, "capIsNew": false }],
    "no-mixed-spaces-and-tabs": [2, "smart-tabs"],
    "no-nested-ternary": 2,
    "no-new-object": 2,
    "no-spaced-func": 1,
    "no-underscore-dangle": 1,
    "padded-blocks": [1, "always"],
    "quotes": [1, "single", "avoid-escape"],
    "semi": [2, "always"],
    "space-before-function-parentheses": [1, "always"],
    "space-after-keywords": [2, "always"],
    "space-before-blocks": [2, "always"],
    "space-in-parens": [1, "never", { "exceptions": ["{}", "[]", "()"] }],
    "space-infix-ops": 2,
    "space-return-throw-case": 2,
    "space-unary-ops": [1, { "words": true, "nonwords": false }],
    "space-in-brackets": [2, "never", {
        "singleValue": false,
        "objectsInArrays": true,
        "arraysInArrays": true,
        "arraysInObjects": true,
        "objectsInObjects": true,
        "propertyName": true
    }],

    "no-var": 1,
    "generator-star": [2, "start"]
  }
}
```

Any help or fixes in case it's purely user error would be much appreciated.

**Comment by btmills** [comment]
What happens if you upgrade to the latest release? We hadn't yet added ES6 class support to `space-before-function-parentheses` in v0.16.2. (Note that we renamed the rule since v0.16.2 to `space-before-function-paren` for consistency with other rules' names, but the configuration is the same except for the shortened name.)

**Comment by nzakas** [comment]
Duplicate of #1989. Already fixed on master.

**Comment by jannisg** [comment]
Upgrading to `0.17.1` fixed this though the rule is still called `"space-before-function-parentheses"` in that version, will probably need to update this on the next release :)

Thanks
